spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/s3/S3AutoConfiguration.java

```diff
@@ -80,17 +87,39 @@ S3ClientBuilder s3ClientBuilder(AwsClientBuilderConfigurer awsClientBuilderConfi
 				connectionDetails.getIfAvailable(), configurer.getIfAvailable(), s3ClientCustomizers.orderedStream(),
 				awsSyncClientCustomizers.orderedStream());
 
+
 		if (ClassUtils.isPresent("software.amazon.awssdk.s3accessgrants.plugin.S3AccessGrantsPlugin", null)) {
 			S3AccessGrantsPlugin s3AccessGrantsPlugin = S3AccessGrantsPlugin.builder()
 					.enableFallback(properties.getPlugin().getEnableFallback()).build();
 			builder.addPlugin(s3AccessGrantsPlugin);
 		}
+
 		Optional.ofNullable(this.properties.getCrossRegionEnabled()).ifPresent(builder::crossRegionAccessEnabled);
 
 		builder.serviceConfiguration(this.properties.toS3Configuration());
 		return builder;
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
+	@ConditionalOnClass(value = S3EncryptionClient.class )
+	S3EncryptionClient.Builder s3EncrpytionClientBuilder(AwsClientBuilderConfigurer awsClientBuilderConfigurer,
```

Comment:
When building a builder twice it will fail due KeyRing which makes sense since this is a secret. Thinking about should we even have it as a bean.

Because of this trying to assert endpoint region is impossible.